File: components/credentials/src/main/java/org/cloudfoundry/credhub/requests/UserGenerateRequest.java
package org.cloudfoundry.credhub.requests;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

public class UserGenerateRequest extends BaseCredentialGenerateRequest {
  @JsonProperty("parameters")
  private StringGenerationParameters passwordGenerationParameters = new StringGenerationParameters();

  @JsonProperty("value")
  private UsernameValue value = new UsernameValue();

  @Override
  @JsonIgnore
  public GenerationParameters getGenerationParameters() {
    if (value.getUsername() != null) {
      passwordGenerationParameters.setUsername(value.getUsername());
    }
    return passwordGenerationParameters;
  }

  public void setGenerationParameters(final StringGenerationParameters generationParameters) {
    passwordGenerationParameters = generationParameters;
  }

  public String getUserName() {
    if (passwordGenerationParameters != null && passwordGenerationParameters.getUsername() != null) {
      return passwordGenerationParameters.getUsername();
    }
    return value.getUsername();
  }

  public void setValue(final UsernameValue value) {
    this.value = value;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/requests/UserGenerateRequest.kt
package org.cloudfoundry.credhub.requests

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty

class UserGenerateRequest : BaseCredentialGenerateRequest() {
    @JsonProperty("parameters")
    private var passwordGenerationParameters: StringGenerationParameters? = StringGenerationParameters()

    @JsonProperty("value")
    private var value = UsernameValue()

    override val generationParameters: GenerationParameters?
        @JsonIgnore
        get() {
            if (value.username != null) {
                passwordGenerationParameters!!.username = value.username
            }
            return passwordGenerationParameters
        }

    val userName: String?
        get() = if (passwordGenerationParameters != null && passwordGenerationParameters!!.username != null) {
            passwordGenerationParameters!!.username
        } else value.username

    fun setGenerationParameters(generationParameters: StringGenerationParameters) {
        passwordGenerationParameters = generationParameters
    }

    fun setValue(value: UsernameValue) {
        this.value = value
    }
}
